scripts/setup_sgtm_tasks_project.py

```diff
@@ -0,0 +1,197 @@
+    client = AsanaClient(pat)
+    client.setup_custom_fields(project_id)
+    client.add_user_to_project(project_id)
+
+
+class AsanaClient(object):
+    def __init__(self, personal_access_token: str):
+        self.client = asana.Client.access_token(personal_access_token)
+        self.client.headers = {"Asana-Enable": "string_ids"}
+        self.workspace_id = self._get_workspace_id()
+
+    def _get_workspace_id(self) -> str:
+        """
+        Get the workspace ID that the user who created the PAT is a part of
+        """
+        pat_user = self.client.users.me()
+        return pat_user["workspaces"][0]["gid"]
+
+    def _get_user_id(self) -> str:
+        """
+        Get the user ID for the user that created the PAT
+        """
+        pat_user = self.client.users.me()
+        return pat_user["gid"]
+
+    def create_project(self, project_name: str, team_id: str) -> str:
+        """
+        Create a project with the given name, shared with the provided team in the workspace
+        """
+        create_project_data = {
+            "name": project_name,
+            "public": True,
+            "team": team_id,
+        }
+        response = self.client.projects.create_project(create_project_data)
+        return response["gid"]
+
+    def setup_custom_fields(self, project_id: str) -> None:
+        """
+        Create the custom fields that SGTM requires and add them to the given project
+        """
+        for custom_field in CUSTOM_FIELDS:
+            custom_field_data = {
+                "name": custom_field.name,
+                "enabled": True,
+                "workspace": self.workspace_id,
+                "resource_subtype": "enum",
+                "enum_options": [],
+            }
+
+            for enum_option in custom_field.enum_options:
+                custom_field_data["enum_options"].append(
+                    {
+                        "name": enum_option.name,
+                        "color": enum_option.color,
+                        "enabled": True,
+                    }
+                )
+
+            response = self.client.custom_fields.create_custom_field(custom_field_data)
+            custom_field_gid = response["gid"]
+
+            self.client.projects.add_custom_field_setting_for_project(
+                project_id,
+                {
+                    "custom_field": custom_field_gid,
+                },
+            )
+
+    def add_user_to_project(self, project_id: str) -> None:
+        """
+        Add the PAT user as a follower to the given project
+        """
+        user_id = self._get_user_id()
+        self.client.projects.add_followers_for_project(
+            project_id, {"followers": user_id}
+        )
+
+
+if __name__ == "__main__":
+    args = parse_args()
+
+    if args.new_project_name and args.existing_project_id:
+        print(
+            "Cannot select both an existing project and a new project, please try again"
+        )
+        sys.exit(0)
```

Comment:
I'd consider leaning into `argparse` to have two subcommands: `update` or `create`, then you wouldn't have to manually check for conflicting options

Reply:
Nice, TIL!